libraries/aws_iam_user.rb

```diff
@@ -2,57 +2,106 @@
 # author: Steffanie Freeman
 # author: Simon Varlow
 # author: Chris Redekop
-
 class AwsIamUser < Inspec.resource(1)
   name 'aws_iam_user'
   desc 'Verifies settings for AWS IAM user'
   example "
-    describe aws_iam_user(name: 'test_user') do
+    describe aws_iam_user(username: 'test_user') do
       it { should have_mfa_enabled }
       it { should_not have_console_password }
     end
   "
-  def initialize(
-    opts,
-    aws_user_provider = AwsIam::UserProvider.new,
-    aws_user_details_provider_ini = AwsIam::UserDetailsProviderInitializer.new,
-    access_key_factory = AwsIamAccessKeyFactory.new
-  )
-    user = opts[:user]
-    user = aws_user_provider.user(opts[:name]) if user.nil?
-    @aws_user_details_provider = aws_user_details_provider_ini.create(user)
-    @access_key_factory = access_key_factory
-  end
 
-  def exists?
-    @aws_user_details_provider.exists?
-  end
+  include AwsResourceMixin
+  attr_reader :username, :has_mfa_enabled, :has_console_password, :access_keys
+  alias has_mfa_enabled? has_mfa_enabled
+  alias has_console_password? has_console_password
 
-  def has_mfa_enabled?
-    @aws_user_details_provider.has_mfa_enabled?
+  def name
+    warn "[DEPRECATION] - Property ':name' is deprecated on the aws_iam_user resource.  Use ':username' instead."
+    username
   end
 
-  def has_console_password?
-    @aws_user_details_provider.has_console_password?
+  def to_s
+    "IAM User #{username}"
   end
 
-  def access_keys
-    @aws_user_details_provider.access_keys.map { |access_key|
-      @access_key_factory.create_access_key(access_key)
-    }
-  end
+  private
 
-  def name
-    @aws_user_details_provider.name
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:username, :aws_user_struct, :name, :user],
+      allowed_scalar_name: :username,
+      allowed_scalar_type: String,
+    )
+    # If someone passed :name, rename it to :username
+    if validated_params.key?(:name)
+      warn "[DEPRECATION] - Resource parameter ':name' is deprecated on the aws_iam_user resource.  Use ':username' instead."
+      validated_params[:username] = validated_params.delete(:name)
+    end
+
+    # If someone passed :user, rename it to :aws_user_struct
+    if validated_params.key?(:user)
+      warn "[DEPRECATION] - Resource parameter ':user' is deprecated on the aws_iam_user resource.  Use ':aws_user_struct' instead."
+      validated_params[:aws_user_struct] = validated_params.delete(:user)
+    end
+
+    if validated_params.empty?
+      raise ArgumentError, 'You must provide a username to aws_iam_user.'
+    end
+    validated_params
   end
 
-  def to_s
-    "IAM User #{name}"
+  def fetch_from_aws
+    backend = BackendFactory.create
+    @aws_user_struct ||= nil # silence unitialized warning
```

Comment:
Thoughts on `unless defined?(aws_user_struct) && aws_user_struct` instead? That wouldn't need a comment explaining it.

Reply:
To me, the comment seems clearer.  @miah?